### PR TITLE
feat(rust): Multiprocessing with Python message processors

### PR DESCRIPTION
**Pull request by untitaker**
Spawn multiple subprocesses in Rust, and run the Python message
processor in each subprocess

By default there is only one process.

### CODE REVIEW

**Review by lynnagara — APPROVED** [review]
Can we add a test?

File: rust_snuba/src/strategies/python.rs [hunk 1]
```diff
@@ -13,131 +17,131 @@ use crate::config::MessageProcessorConfig;
...
-                self.message_carried_over = Some(transformed_message);
+    fn check_for_results(&mut self) {
+        // procspawn has no join() timeout that does not consume the handle on timeout.
+        //
+        // Additionally we have observed, at least on MacOS, that procspawn's active_count() only
+        // decreases when the handle is consumed. Therefore our count of actually saturated
+        // processes is `self.processing_pool.active_count() - self.handles.len()`.
+        //
+        // If no process is saturated (i.e. above equation is <= 0), we can conclude that all tasks
+        // are done and all handles can be joined and consumed without waiting.
```
Comment (untitaker) [inline rust_snuba/src/strategies/python.rs:56]: @mitsuhiko does this reasoning make sense to you? I am not sure what the intended behavior of `active_count` is

Comment (mitsuhiko) [reply rust_snuba/src/strategies/python.rs:56]: I think that logic makes sense but honestly this probably needs changing in procspawn.

**Comment by lynnagara** [comment]
Consumer seems to be backlogging, reverting..

**Pull request merged**